django_app/redbox_app/redbox_core/views.py

```diff
@@ -153,16 +161,56 @@ def remove_doc_view(request, doc_id: str):
 
 @login_required
 def sessions_view(request, session_id: str = ""):
-    # Hard-coding some chat history for now
-    session_history = [
-        {"session_id": "chat1", "title": "Chat 1", "url": "/sessions/chat1"},
-        {"session_id": "chat2", "title": "Chat 2", "url": "/sessions/chat2"},
-    ]
+    chat_history = models.ChatHistory.objects.all().filter(users=request.user)
+
+    messages = []
+    if session_id:
+        current_chat = models.ChatHistory.objects.get(id=session_id)
+        messages = models.ChatMessage.objects.all().filter(chat_history=current_chat)
 
-    context = {"session_history": session_history}
+    context = {
+        "session_id": session_id,
+        "messages": messages,
+        "chat_history": chat_history,
+    }
 
     return render(
         request,
         template_name="sessions.html",
         context=context,
     )
+
+
+@require_http_methods(["POST"])
+def post_message(request: HttpRequest) -> HttpResponse:
+    message_text = request.POST.get("message", "New chat")
+
+    # get current session, or create a new one
```

Comment:
not a blocker, but im wondering if you should make this distinction with PUT vs POST?

Reply:
Interesting. Let's discuss this. @KevinEtchells?

Reply:
I used POST to match the core-api documentation, but happy with either.

Reply:
It might be better from a pure REST point of view to consider a message to be a sub resource of the session, so a new chat would be a post to `/session/` and a message to an existing chat would be a post to `/session/12345/message/`, but I'm not inclined to worry about it too much unless it starts giving us an issue.